### PR TITLE
DEV: combines like_users & reaction_users both if reaction is main

### PR DESCRIPTION
**Pull request by Ahmedgagan**
@jjaffeux combines like_users & reaction_users both if reaction is main in `post_reactions_users` / `reactions-users` route

### CODE REVIEW

**Comment by jjaffeux** [comment]
Please write a spec exhibiting what behavior is this changing/fixing

**Comment by Ahmedgagan** [comment]
> Please write a spec exhibiting what behavior is this changing/fixing

Sure, I'll write a spec for this behavior

**Comment by Ahmedgagan** [comment]
@jjaffeux Added the rspec for this behavior

**Comment by jjaffeux** [comment]
Is my fix in the serializer still need with this?

**Comment by Ahmedgagan** [comment]
Yes both are required, as we call this route only when anyone hover's on panel OR open the panel

File: app/controllers/discourse_reactions/custom_reactions_controller.rb [hunk 1]
```diff
@@ -77,15 +77,26 @@ def post_reactions_users
       likes = post.post_actions.where('deleted_at IS NULL AND post_action_type_id = ?', PostActionType.types[:like]) if !reaction_value || reaction_value == DiscourseReactions::Reaction.main_reaction_id
 
       if likes.present?
+        main_reaction = DiscourseReactions::Reaction.find_by(reaction_value: DiscourseReactions::Reaction.main_reaction_id, post_id: post.id)
+        count = likes.length.to_i
```
Comment (jjaffeux) [inline app/controllers/discourse_reactions/custom_reactions_controller.rb]: why do we need to_i  here ? can length be something else than integer ?

Comment (Ahmedgagan) [reply app/controllers/discourse_reactions/custom_reactions_controller.rb]: No😅 nice catch Ill change it

File: spec/requests/custom_reactions_controller_spec.rb [hunk 2]
```diff
@@ -155,6 +155,21 @@
     end
   end
 
+  context 'changing discourse_reactions_like_icon' do
+    before do
+      SiteSetting.discourse_reactions_reaction_for_like = 'laughing'
+    end
+
+    it 'merges identic custom reaction into likes' do
+      user_5 = Fabricate(:user)
```
Comment (jjaffeux) [inline spec/requests/custom_reactions_controller_spec.rb]: why user 5 ? can you use let for this ?

File: spec/requests/custom_reactions_controller_spec.rb [hunk 3]
```diff
@@ -155,6 +155,21 @@
...
+      like = Fabricate(:post_action, post: post_2, user: user_5, post_action_type_id: PostActionType.types[:like])
```
Comment (jjaffeux) [inline spec/requests/custom_reactions_controller_spec.rb]: same here

File: spec/requests/custom_reactions_controller_spec.rb [hunk 4]
```diff
@@ -155,6 +155,21 @@
     end
   end
 
+  context 'changing discourse_reactions_like_icon' do
```
Comment (jjaffeux) [inline spec/requests/custom_reactions_controller_spec.rb]: are we changing the icon or the reaction? Be very careful with wording please

Comment (Ahmedgagan) [reply spec/requests/custom_reactions_controller_spec.rb]: oooh my bad we are changing `reaction`

**Comment by Ahmedgagan** [comment]
@jjaffeux used `fab!` instead of `user_5=` & changed the context name in the latest commit

File: spec/requests/custom_reactions_controller_spec.rb [hunk 5]
```diff
@@ -155,6 +155,21 @@
     end
   end
 
+  context 'changing discourse_reactions_reaction_for_like' do
```
Comment (jjaffeux) [inline spec/requests/custom_reactions_controller_spec.rb]: this test still doesn’t make it clear what's happening, do a before/after, eg:

initial situation assert X likes Y laughing

change setting

assert X likes and Y laughing

File: spec/requests/custom_reactions_controller_spec.rb [hunk 6]
```diff
@@ -153,6 +155,24 @@
       get "/discourse-reactions/posts/1000000/reactions-users.json?reaction_value=test"
       expect(response.status).to eq(400)
     end
+
+    it 'merges identic custom reaction into likes' do
```
Comment (Ahmedgagan) [inline spec/requests/custom_reactions_controller_spec.rb:159]: @jjaffeux How does it look now?

**Pull request merged**